fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -115,312 +147,241 @@ def method(self):
 
 
 class LeakySplits(fob.BrainMethod):
-    def initialize(self, samples, brain_key=None):
-        self.index = LeakySplitsIndex(samples, self.config, brain_key)
-        return self.index
+    def initialize(self, samples, similarity_index, split_views):
+        return LeakySplitsIndex(
+            samples, self.config, similarity_index, split_views
+        )
+
+    def get_fields(self, samples, _):
+        fields = []
+        if self.config.embeddings_field is not None:
+            fields.append(self.config.embeddings_field)
 
-    def cleanup(self, samples, brain_key):
-        self.index._similarity_method.cleanup(samples, brain_key)
-        super.cleanup(samples, brain_key)
+        return fields
 
 
 class LeakySplitsIndex(fob.BrainResults):
-    def __init__(self, samples, config, brain_key):
-        super().__init__(samples, config, brain_key)
+    def __init__(self, samples, config, similarity_index, split_views):
+        super().__init__(samples, config, None)
```

Comment:
If we wanted to add support for `compute_leaky_splits(..., brain_key=...)`, then this class would need to be refactored such that this would work:

```py
index = dataset.load_brain_results(brain_key)
index.find_leaks()
```

That would be a reasonable thing to do **IF** the user originally provided a `similarity_index`, as we could just use `config.similarity_index` to load it.

If the user didn't provide a similarity index, then they would need to have provided either an `embeddings="field_name"` or `model="model_name"` so that we could load/recompute the necessary embeddings. However, by convention, `load_brain_results()` should really not do any *novel* compute, it is assumed that any expensive compute was done when the method was originally called and, as per the `BrainResults` class naming, the **resulting** data should be stored such that it can be loaded **quickly** via `load_brain_results()` in a new session.

The TLDR here is that `compute_similarity()` is really doing all the heavy-lifting here. `compute_leaky_splits()` is really just a function that you call that uses a similarity index to do something useful.